### ISSUE TITLE
failure on "Set the user token and refresh token time to same time length"

### ISSUE THREAD

**Issue by darkylmnx** [websanova/vue-auth#121]
hi,

i followed the step and "Set the user token and refresh token time to same time length" in my Laravel backend using JWT https://github.com/tymondesigns/jwt-auth

But the thing is, now if the device get's lost with the use logged in inside, the "founder" / "thief" of the device can now access the device and use an old token since the refresh token has the same expire data as the normal token and since the normal token won't be revoked when the refresh token will be issued.

I'd like a clear explanation on how JWT or this Vue lib handle's well this case. thanks

**Comment by websanova** [comment]
lol, I think there is plenty of documentation online for this. Not sure why
you are asking for it here.

BUT, in short if someone has access to your phone/app it wouldn't matter
what the timeout is. As long as they are active with it it will constantly
refresh. It makes no difference whether it's a thief or you.

The idea of the token is just to try to provide some kind of failsafe to
not leave an active token their too too long. Generally it's​ about two
weeks. But maybe a lot shorter if you need tighter security.

Just think about it with something like FB. If someone has access to your
phone is it gonna ask them for a password? No. Only after few weeks of
inactivity (on that particular device).

I think where people get confused with JWT is because  of the two
tokens/timeouts. There is a short lived one and a long lived one. The long
lived one is usually about two weeks and is called your refresh token. The
short one is the one that maybe is every hour.

Basically, it's the long lived one is the one that gives access not the
short one. As long as the refresh one is set to two weeks you will keep
getting tokens (also where grace period for expired tokens comes into
play). The reason for this is just security on the back end. So generally
the refresh token can be stored in a more central secure area.

If it's all on the same server, it's quite redundant, bit either way it
should work.

Anyway, do your own research ;)

**Comment by websanova** [comment]
btw, the plugin doesn't force you to use the same times. You can make the
short lived one shorter if you like ;)

**Comment by darkylmnx** [comment]
i'm sorry maybe be i was miseunderstood, my english is not perfect yet. What i was saying was : 
You give us 3 options : 
- Set a new token with each request (with old tokens living slightly longer in case of async requests)
- Set a timer or counter for a token refresh request
- Check the tokens expiration date before or after each request
- Set the user token and refresh token time to same time length and call refresh with each reload of the app

and said that the last one was the default behaviour and what i was asking was why compared to the others because it has the same issues, since if the two has the same time there's no point having two refreshable tokens in the first place.

as said here : 

> **Token time to live - ttl**
> 
> This is the length of time, in minutes, that your token will be considered valid. It is recommended that this is kept as short as possible, especially if utilising token refreshing.
> 
> **Refresh time to live - refresh_ttl**
> 
> This is the length of time, in minutes, that you can refresh a token within. For example, if you set this to 2 weeks, then you will only be able to refresh the same chain of tokens for a maximum of 2 weeks before the token will be 'un-refreshable' and the result will always be a TokenExpiredException. So after this time has passed, a brand new token must be generated, and usually that means the user has to login again. 

Anyway i think for now i will just do simple and leave both like you said. I'll take time to deeply search more about the workflow.
